src/tokenizer/index.js

```diff
@@ -555,6 +572,23 @@ export default class Tokenizer extends LocationParser {
     for (let i = 0, e = len == null ? Infinity : len; i < e; ++i) {
       const code = this.input.charCodeAt(this.state.pos);
       let val;
+
+      if (this.hasPlugin("numericSeparator")) {
+        const prev = this.input.charCodeAt(this.state.pos - 1);
+        const next = this.input.charCodeAt(this.state.pos + 1);
+        if (code === 95) {
+          if (forbiddenNumericLiteralSeparatorSiblings.includes(prev) ||
+              forbiddenNumericLiteralSeparatorSiblings.includes(next) ||
+              Number.isNaN(next)) {
+            this.raise(this.state.pos, "Invalid NumericLiteralSeparator");
```

Comment:
I might actually create a different error message for multi `_`, if it's at the end or the beginning, or around invalid chars but we can do that later too.